stage1/init/kvm/mount.go

```diff
@@ -147,36 +149,51 @@ func PodToSystemdHostMountUnits(root string, volumes []types.Volume, appNames []
 
 // AppToSystemdMountUnits prepare bind mount unit for empty or host kind mounting
 // between stage1 rootfs and chrooted filesystem for application
-func AppToSystemdMountUnits(root string, appName types.ACName, mountPoints []types.MountPoint, unitsDir string) error {
+func AppToSystemdMountUnits(root string, appName types.ACName, volumes []types.Volume, ra *schema.RuntimeApp, unitsDir string) error {
+	app := ra.App
 
-	for _, mountPoint := range mountPoints {
+	vols := make(map[types.ACName]types.Volume)
+	for _, v := range volumes {
+		vols[v.Name] = v
+	}
+
+	mounts, err := s1common.GenerateMounts(ra, vols)
+	if err != nil {
+		return err
+	}
+
+	for _, m := range mounts {
+		vol := vols[m.Volume]
 
-		name := mountPoint.Name.String()
 		// source relative to stage1 rootfs to relative pod root
-		whatPath := filepath.Join(stage1MntDir, name)
+		whatPath := filepath.Join(stage1MntDir, vol.Name.String())
 		whatFullPath := filepath.Join(root, whatPath)
 
 		// destination relative to stage1 rootfs and relative to pod root
-		wherePath := filepath.Join(common.RelAppRootfsPath(appName), mountPoint.Path)
+		wherePath := filepath.Join(common.RelAppRootfsPath(appName), m.Path)
 		whereFullPath := filepath.Join(root, wherePath)
 
 		// readOnly
+		readOnly := s1common.IsMPReadOnly(app.MountPoints, vol.Name)
```

Comment:
Maybe put it into `else` part of an `if` below? The `readOnly` variable get clobbered if `vol.ReadOnly` is not `nil`.